cmd/root.go

```diff
@@ -68,3 +82,81 @@ Commit info: {{index .Annotations "commitInfo"}}
 }
 
 func init() {}
+
+// checkGitHashMismatch will compare the hash of the git repository vs the git
+// hash the ghpc binary was compiled against
+func checkGitHashMismatch() (string, string, string) {
+	if len(GitCommitHash) == 0 {
+		return "", "", ""
+	}
+
+	repo, dir, err := hpcToolkitRepo()
+	if err != nil {
+		return "", "", ""
+	}
+
+	head, err := repo.Head()
+	if err != nil {
+		return "", "", ""
+	}
+
+	if GitCommitHash != head.Hash().String() {
+		b := head.Name().Short()
+		h := head.Hash().String()
+		return b, h, dir
+	}
+	return "", "", ""
+}
+
+// hpcToolkitRepo will find the path of the directory containing the hpc-toolkit
+// starting with the working directory and evaluating the parent directories until
+// the toolkit repository is found. If the HPC Toolkit repository is not found by
+// traversing the path, then the executable directory is checked.
+func hpcToolkitRepo() (*git.Repository, string, error) {
```

Comment:
What case is all of this protecting against?   
1. If I call `git rev-list --max-parents=0 HEAD` in a subdirectory of a git repo it still works just fine. For example I get the same  response if I call it from `hpc-toolkit/` or `hpc-toolkit/examples/`. I would assume `git.PlainOpen(dir)` would work even if you are in a subdir (I could be wrong). The only thing I can think that we are protecting against by crawling up the dir tree is if there is a git repo inside of the hpc-toolkit repo. This is possible but I don't think it is a requirement and I don't know if the extra complexity is worth it.
2. I don't think we should ever be falling back to the executable directory. I tried to bring this up in [this comment](https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/618#discussion_r993661237). Let's discuss.

Reply:
`git.PlainOpen(dir)` expects `dir` to be the root of the repo and to contain the `.git` directory, so I initially put this to walk the parent directories' path until the root of the git repo is identified. 

Following your comment (thanks!!), I looked again and found I can instead do something like the below, and rely on the library to find the `.git` directory.

```
o := git.PlainOpenOptions{DetectDotGit: true}
repo, _ := git.PlainOpenWithOptions(repoDir, &o)
```

With that change, the only benefit of this block would be if ./ghpc is called from within a subfolder that contains a different git repo.

I'm leaving this open and unchanged for now, pending our discussion.